### DISCUSSION TITLE
Music genre recognition

### DISCUSSION THREAD

**Discussion by marcelklehr** [nextcloud/recognize#108] [Announcements]
Hello there!

We're currently testing a new feature: Music genre recognition: :musical_note: 

You may have had the problem trying to categorize your music collection before. Sadly most genre labels for music are utterly useless: What's indie rock to some is called alternative to others, most labels are only available on an album-level anyway, and some people go deep down the rabbit hole to create genres that are so specific that they will only ever match one band ("ambient doom jazz", anyone?)

We built a music model that recognizes the 15 top genres based directly on audio features and adds the corresponding nextcloud tags to the classified files. It's not perfect but it's nice to create playlists of songs that are somewhat similar. You can listen to your songs categorized by recognize using the audio player app.

Try the beta over here: https://github.com/marcelklehr/recognize/releases/tag/v1.7.0-beta.1

If you have problems, critique, feedback, feel free to leave a comment below.

**Comment by IlyaER** [comment]
Hello,
I've tried the music feature and it seems very intresting.
There is a strange thing however as most of my music is recognized in a completely wrong way.
This may be connected with the fact that my music is mostly classical and the model was trained on completely different popular music.
E.g.:
O Holy Night.mp3 = ["country","new age"]
Solveig's Song.mp3 = ["jazz"]
Whereas both of them should definitely be classified as classical music.
What can be done to improve the recognition?

**Reply by marcelklehr** [reply]
> This may be connected with the fact that my music is mostly classical and the model was trained on completely different popular music.

Yes, that could be the reason.

> What can be done to improve the recognition?

A new model would need to be trained with more diverse samples. I still intend to publish my model code, so while I may not have time to train a new model other's may pick up the slack and do it with the help of the code.
